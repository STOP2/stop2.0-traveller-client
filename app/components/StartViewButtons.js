import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Text, ListView, View, TouchableOpacity } from 'react-native'
import { Actions } from 'react-native-router-flux'

import styles from '../styles/stylesheet'
import strings from '../resources/translations'

class StartViewButtons extends Component{
  constructor(props) {
    super(props)

    this.state = {

    }

  }
  render() {
    const goToBusListView = () => Actions.departures()
    return (
    <View style={styles.start}>
    <TouchableOpacity onPress={goToBusListView}>
    <Text style={styles.startText}>{strings.onStop}</Text>
    </TouchableOpacity>
    <TouchableOpacity onPress={goToBusListView}>
    <Text style={styles.startText}>{strings.onBus}</Text>
    </TouchableOpacity>
    </View>)
  }
}

export default StartViewButtons;
