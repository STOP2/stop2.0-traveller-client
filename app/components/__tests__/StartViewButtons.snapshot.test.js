import 'react-native'
import React from 'react'
import StartViewButtons from '../StartViewButtons'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('StartViewButtons component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<StartViewButtons />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
