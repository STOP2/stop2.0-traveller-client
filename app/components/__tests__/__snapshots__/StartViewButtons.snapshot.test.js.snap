exports[`StartViewButtons component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 5,
    }
  }>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="At the stop"
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00B9E4",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginBottom": 5,
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "gotham-rounded-medium",
          },
          Object {
            "color": "white",
            "fontSize": 50,
          },
        ]
      }>
      At the stop
    </Text>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel="On the bus"
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F092CD",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginTop": 5,
        "opacity": 1,
      }
    }
    testID={undefined}>
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 90,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "gotham-rounded-medium",
          },
          Object {
            "color": "white",
            "fontSize": 50,
          },
        ]
      }>
      On the bus
    </Text>
  </View>
</View>
`;
